src/components/profile/profile.tsx

```diff
@@ -13,6 +14,7 @@ import { goToUrl } from "../../store/routerHistory";
 import "./profile.scss";
 
 const mapStateToProps = (state: IRootState) => ({
+    currentState: state.state.currentState,
```

Comment:
```suggestion
    currentState: state.currentState,
```
Shouldn't this be ^?

Reply:
Object `state: IRootState` contains `state: IStateState` so `state.state.currentState` is correct. But I agree the naming is very confusing (especially `IStateState`). Any suggestions for a different name for the state objects?

Reply:
@Juuldamen did you come up with `IStateState` or did @roelvdboom ?
I'm probably missing the reasoning behind its name, could you enlighten @roelvdboom ?

Reply:
I followed the other naming conventions (eg. `IAuthenticationState` and `ITransactionState`) by @roelvdboom.  But that doesn't really work in this case. What do you think we could change to this naming convention @roelvdboom?

Reply:
How about `IPersonalWalletState` @Juuldamen ?